### PR TITLE
Property-based GLM test

### PR DESCRIPTION
**Pull request by michaelweilsalesforce**
**Related issues**
Adding more Tests for our ML capabilities (TDD)

**Describe the proposed solution**
Property based tests to check each of our the Regression Models (Linear Regression, Random Forest, GLM) will be picked based on the nature of the data (and response) generated

**Describe alternatives you've considered**
Doing the same tests on one single dataset instead

### CODE REVIEW

**Comment by tovbinm** [comment]
@michaelweilsalesforce please update PR description. also, is it ready for review?

File: core/src/main/scala/com/salesforce/op/utils/stages/FitStagesUtil.scala [hunk 1]
```diff
@@ -262,6 +261,7 @@ private[op] case object FitStagesUtil {
     val stages = stagesLayer.map(_._1)
     val (estimators, noFit) = stages.partition(_.isInstanceOf[Estimator[_]])
     val fitEstimators = estimators.map { case e: Estimator[_] =>
+      println(e.getInputFeatures().toSeq)
```
Comment (tovbinm) [inline core/src/main/scala/com/salesforce/op/utils/stages/FitStagesUtil.scala]: remove println

**Comment by michaelweilsalesforce** [comment]
@Jauntbox These tests are actually painful for our runtime (~15-20 mins/ unit test)

**Comment by tovbinm** [comment]
Builds will be killed if not output is produced for 10 minutes, so we can’t have an individual test take longer than that.

**Pull request merged**